### DISCUSSION TITLE
Can this action be modified to work with other repository hosts?

### DISCUSSION THREAD

**Discussion by Andre601** [lowlighter/metrics#1506] [Q&A]
I have an account on codeberg.org and they have updated to a new Forgejo (Soft-fork of Gitea) version that implements user readmes...

As of right now do I simply use links to the generated SVG files in my GitHub Profile Readme in that other readme file to display something, but I would like to change that.
I preferably would like to display statistics related to the codeberg site, but I cannot think of anything that could provide such a thing...

So I wonder if I can use metrics with some customization/tweaks to use on Codeberg itself.
They unfortunately do not have Actions support active (It's available but due to security concerns and lack of resources is it not enabled) and their alternative which is Woodpecker-CI wouldn't allow metrics to be used like that... At least not that I know of any way that is...

So... My idea is to simply have a repo here on GitHub with GitHub actions executing metrics, so that SVGs are generated which I would then use on Codeberg... I hope this sentence makes sense...

The question now is, what possible changes, if any, would be needed to make metrics work for codeberg. As mentioned does it use Forgejo, which is a soft-fork of Gitea, so API endpoints are available and should be the same as with Gitea itself... J-just don't ask me which one is used (I assume it follows a similar pattern to GitHub API paths tho).

**Comment by Vault108** [comment]
It would most likely have to be fully rewritten for it to work with codeberg. It wont be a "quick" fix. I just peaked at source for Metrics on mobile and it looks like its querying the GraphQL API to get the data from Git.
